include/picongpu/plugins/radiation/Radiation.hpp

```diff
@@ -664,396 +659,196 @@ namespace picongpu
-                            currentStep,
-                            radSplashType,
-                            (meshesPathName + dataLabelsDetectorDirection(detectorDim)).c_str(),
-                            "unitSI",
-                            &factorDirection);
-
-                        /* position */
-                        std::vector<float_X> positionMesh(simDim, 0.0); /* there is no offset - zero */
-                        hdf5DataFile.writeAttribute(
-                            currentStep,
-                            splashFloatXType,
-                            (meshesPathName + dataLabelsDetectorDirection(detectorDim)).c_str(),
-                            "position",
-                            1u,
-                            splash::Dimensions(simDim, 0, 0),
-                            &(*positionMesh.begin()));
+                            // select data
+                            for(int copyIndex = 0; copyIndex < N_tmpBuffer; ++copyIndex)
+                            {
+                                tmpBuffer[copyIndex]
+                                    = ((picongpu::float_64*)
+                                           values.data())[ampIndex + Amplitude::numComponents * copyIndex];
+                            }
+                            // write actual data
+                            mesh_amp[dir].storeChunk(tmpBuffer, offset_amp, extent_amp);
+                            openPMDdataFile.flush();
+                        }
                     }
+                    // end: write amplitude data
+
+
+                    // start: write observer
+                    // prepare n meshes
+                    ::openPMD::Mesh mesh_n = openPMDdataFileIteration.meshes[dataLabelsDetectorDirection(-1)];
+                    // write mesh attributes
+                    mesh_n.setGeometry(::openPMD::Mesh::Geometry::cartesian); // set be default
+                    // mesh_n.setGeometryParameters( geometryParameters );
+                    mesh_n.setDataOrder(::openPMD::Mesh::DataOrder::C);
+                    mesh_n.setGridSpacing(std::vector<double>{1.0, 1.0, 1.0});
+                    mesh_n.setGridGlobalOffset(std::vector<double>{0.0, 0.0, 0.0});
+                    mesh_n.setGridUnitSI(1.0);
+                    mesh_n.setAxisLabels(std::vector<std::string>{"detector direction index", "None", "None"});
+                    mesh_n.setUnitDimension(std::map<::openPMD::UnitDimension, double>{});
 
+                    {
+                        std::vector<float_64> tmpBuffer(parameters::N_observer);
 
-                    /* save detector frequencies */
-                    splash::Dimensions bufferSizeOmega(1, radiation_frequencies::N_omega, 1);
+                        // reshape abstract MeshRecordComponent
+                        ::openPMD::Datatype datatype_n = ::openPMD::determineDatatype<float_64>();
+                        ::openPMD::Extent extent_n = {parameters::N_observer, 1, 1};
```

Comment:
This should be `{1, 1, parameters::N_observer};` to be `C` order.

Reply:
You can have a look into our openPMD IO plugin, N-dimensional PIConGPU data extents must be transposed.

Reply:
I am not sure. The current code seems to match the `axisLabels` set later. For N-1-1 or 1-1-N arrays the data order does not change anything.

Reply:
I am also not 100% sure but thought if it is a 3Dimensional data object where only one dimension is >1 the last should be the set value if it is C-ordering.
Never the less I have no problem if we keep it how it is.

Reply:
> if it is a 3Dimensional data object where only one dimension is >1 the last should be the set value if it is C-ordering.

This is not true

Reply:
@sbastrakov is right: the strange order is used to have a consistent data dimensionality between all datasets. 
It is intended to have `{parameters::N_observer, 1, 1}` because the mathematically first (C-like last) axis is in all data sets the observer axis. This looks strange for the frequency and observer datasets because they are both 1D, but in combination with the amplitude data sets this is consistent and does not need explanation which axis matches which dimension in the amplitude data set (in case there is the same number of observers and frequencies.) 

This should be keep due to two reasons:
 - backward compatibility (that will be changed with #3690)
 - "parallel" dimensions should be enforced until the openPMD standard supports not only `axisLabels` and equidistant `gridSpacing` but arbitrary values as well (that's why the axis points are  mapped the way they ae right now)